src/org/parosproxy/paros/core/scanner/VariantHeader.java

```diff
@@ -32,15 +37,37 @@
  * @author andy
  */
 public class VariantHeader implements Variant {
+	
+	//might still be publicly used.
+	@Deprecated 
+	public static final String[] injectableHeaders = {
+		  HttpRequestHeader.USER_AGENT,
+		  HttpRequestHeader.REFERER,
+		  HttpRequestHeader.HOST
+	};
 
-    // I've found an XSS using this payload on the Host header
-    // "%s-->\">'>'\"<sfi%06uv%06u>"
-
-    public static final String[] injectableHeaders = {
-        HttpRequestHeader.USER_AGENT,
-        HttpRequestHeader.REFERER,
-        HttpRequestHeader.HOST
+	//headers converted to lowercase to make comparison easier later.    
+    private static final String [] injectablesTempArray = {    	

```

Comment:
Since the array is now just used to initialise the `HashSet`, it could be done in a static block to let it go way (not that it would make much difference).